### PR TITLE
Added blacklist logic and refactored some code

### PR DESCRIPTION
**Pull request by AliceYuan**
added back CORS
added blacklist for checking for website requests 
added unit tests

### CODE REVIEW

File: client/api.py [hunk 1]
```diff
@@ -4,18 +4,21 @@
 
 from flask import Flask
 from flask import request
+from flask.ext.cors import CORS
```
Comment (MeenuV) [inline client/api.py]: I thought we were removing CORS?

Comment (AliceYuan) [inline client/api.py]: I realized the issue I was having with gunicorn was some venv issue so I put it back in

File: client/blacklist.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+from netaddr import IPNetwork, IPAddress
+from netaddr.core import AddrFormatError
+
+
+bl_request_ip = []  # array of tuples (network mask, port)
+bl_website_ip = []  # array of tuples (network mask, port)
+
+
+def build_inc_request_blacklist(logger):
+    with open("config/blacklist_inc_request_ip.txt") as f:
```
Comment (MeenuV) [inline client/blacklist.py]: Should we give some flexibility as to where they want to store this information? They might not want to push a file containing their blacklists. Maybe accept the file name as a config parameter instead or something?

Comment (AliceYuan) [inline client/blacklist.py]: I think this is not a big deal, they should be using our config files anyway

**Comment by s-varun** [comment]
+1

**Comment by samiyaakhtar** [comment]
r+

**Pull request merged**